### ISSUE TITLE
Bug: [no-unnecessary-template-expression] do not flag multiline template strings that interpolate trailing whitespace

### ISSUE THREAD

**Issue by kirkwaiblinger** [typescript-eslint/typescript-eslint#10360]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.6.2&fileType=.tsx&code=MYewdgzgLgBAZiEMC8MAGAoGMoAsCWEMuAhkVAE4n4A2%2BYA5jAO4FQCmEADicOwFwASAN4BybNlEBfDGgDcQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tiacTJTIAhtEK18iALbF4QifQAexaIMocm6KAHcRG8GAC%2BIA0A&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
const foo = `
  this has trailing whitespace:${'    '}
`;
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-unnecessary-template-expression": "warn"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

This case should be allowed since it's a useful way of explicitly showing trailing whitespace

### Actual Result

This code is flagged by the rule.

### Additional Info

It's also really annoying to disable the report. You can't use a next-line disable, since the preceeding line is part of the template string.

Inspired by https://github.com/typescript-eslint/typescript-eslint/pull/10244#discussion_r1826953917
cc @bradzacher
